haigha/connections/rabbit_connection.py

```diff
@@ -208,6 +240,91 @@ def _recv_nack(self, method_frame):
                 self._last_ack_id = delivery_tag
                 self._nack_listener(self._last_ack_id, requeue)
 
+    def consume(self, *args, **kwargs):
```

Comment:
This overrides `BasicClass.consume()` and adds a new kwarg `cancel_cb` for notifying user when a consumer is cancelled by the broker.